consensus/service/src/timestamp_validator.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+use displaydoc::Display;
+
+/// Provides logic for validating and ensuring that a timestamp exists when
+/// combining `ConsensusValues`.
+
+const MAX_TIMESTAMP_AGE: u64 = 30 * 1000; // 30 seconds
+
+/// Errors related to validating timestamps
+#[derive(Debug, Display, Clone, PartialEq)]
+pub enum Error {
+    /// The timestamp is in the future now{0}, timestamp{1}
+    InFuture(u64, u64),
+    /** The timestamp is older than the last bock: last_bock timestamp{0},
+    timestamp{1} */
```

Comment:
`timestamp{0}` seems off, should it just be `last_block:{0} cur_block:{1}` or something like that?

Reply:
conversation brought over to, https://github.com/mobilecoinfoundation/mobilecoin/pull/3733/files#r1399804585